homeassistant/components/ezviz/select.py

```diff
@@ -46,45 +138,43 @@ async def async_setup_entry(
     ]
 
     async_add_entities(
-        EzvizSelect(coordinator, camera)
+        EzvizSelect(coordinator, camera, ALARM_SOUND_MODE_SELECT_TYPE)
         for camera in coordinator.data
         for switch in coordinator.data[camera]["switches"]
-        if switch == SELECT_TYPE.supported_switch
+        if switch == ALARM_SOUND_MODE_SELECT_TYPE.supported_switch
+    )
+
+    async_add_entities(
+        EzvizSelect(coordinator, camera, BATTERY_WORK_MODE_SELECT_TYPE)
+        for camera in coordinator.data
+        if coordinator.data[camera]["device_category"]
+        == DeviceCatagories.BATTERY_CAMERA_DEVICE_CATEGORY.value
     )
```

Comment:
Please add all the entities you want to add in a list and execute `async_add_entities` once, since its expensive